File: 2022/Day1/part_one.py
import sys

def main():
    with open(sys.argv[1], "r") as f:
        lines = f.readlines()
        maximum = 0
        current = 0
        for line in lines:
            line = line.strip()
            if line == "":
                maximum = max(maximum, current)
                current = 0
            else:
                current = current + int(line)
        print(maximum)

if __name__ == "__main__":
    main()

File: 2022/Day1/part_one.js
const fs = require("fs");

const fileName = process.argv[2] || "data/input.txt";
const content = fs.readFileSync(fileName, "utf8");
const lines = content.split(/\r?\n/);

const result = lines.reduce(
  (acc, line) => {
    if (line === "") {
      const max = Math.max(acc.max, acc.current);
      return { max, current: 0 };
    }
    const value = parseInt(line, 10);
    return { max: acc.max, current: acc.current + value };
  },
  { max: 0, current: 0 }
);

console.log(result.max);
